build_openroad.sh

```diff
@@ -18,146 +52,306 @@ Options:
+                        shift
+                        ;;
+                --no_init)
+                        OPENROAD_FLOW_NO_GIT_INIT=1
+                        ;;
+                -t|--threads)
+                        PROC="$2"
+                        shift
+                        ;;
+                -n|--nice)
+                        NICE="nice"
+                        ;;
+                -c|--copy-platforms)
+                        DOCKER_COPY_PLATFORMS="YES"
+                        ;;
+                --docker-args-overwrite)
+                        DOCKER_OVERWIRTE_ARGS="YES"
+                        ;;
+                --docker-args)
+                        DOCKER_USER_ARGS="$2"
+                        shift
+                        ;;
+                --yosys-args-overwrite)
+                        YOSYS_OVERWIRTE_ARGS="YES"
+                        ;;
+                --yosys-args)
+                        YOSYS_USER_ARGS="$2"
+                        shift
+                        ;;
+                --openroad-args-overwrite)
+                        OPENROAD_APP_OVERWIRTE_ARGS="YES"
+                        ;;
+                --openroad-args)
+                        OPENROAD_APP_USER_ARGS="$2"
+                        shift
+                        ;;
+                --lsoracle-args-overwrite)
+                        LSORACLE_OVERWIRTE_ARGS="YES"
+                        ;;
+                --lsoracle-args)
+                        LSORACLE_USER_ARGS="$2"
+                        shift
+                        ;;
+                --install-path)
+                        INSTALL_PATH="$2"
+                        shift
+                        ;;
+                --clean)
+                        CLEAN_BEFORE="YES"
+                        ;;
+                --clean-force)
+                        CLEAN_BEFORE="YES"
+                        CLEAN_FORCE="YES"
+                        ;;
+                -*|--*) # unsupported flags
+                        echo "[ERROR FLW-0005] Unsupported flag $1." >&2
+                        usage 2> /dev/null
+                        exit 1
+                        ;;
+        esac
+        shift
 done
 
-# Choose install method
-if [ -z ${BUILD_METHOD+x} ] && command -v docker &> /dev/null; then
-  echo "[INFO FLW-0000] Using docker build method. This will create a docker image tagged '${DOCKER_TAG}'."
-  build_method="DOCKER"
+if [[ "${DOCKER_OVERWIRTE_ARGS}" == "YES" ]]; then
+        DOCKER_ARGS="${DOCKER_USER_ARGS}"
 else
-  echo "[INFO FLW-0000] Using local build method. This will create binaries at 'tools/install/..'."
-  build_method="LOCAL"
+        DOCKER_ARGS="${DOCKER_ARGS} ${DOCKER_USER_ARGS}"
 fi
 
-# Clone repositories
-if [ -z ${OPENROAD_FLOW_NO_GIT_INIT+x} ]; then
-  git submodule update --init --recursive
+YOSYS_ARGS="\
+CONFIG=gcc \
+PREFIX=${INSTALL_PATH}/yosys \
+ABCREV=bafd2a7 ABCURL=https://github.com/berkeley-abc/abc \
+"
```

Comment:
These variables depend on INSTALL_PATH.
INSTALL_PATH needs to be defined before parsing arguments so it can be updated.
If we define these variables before parsing the arguments they will not get the possibly updated INSTALL_PATH.

I guess we can define before and then just append the INSTALL_PATH dependent after, but I thought this was better. What do you think: define before and append or define after?

Reply:
Fixed on f8863c1